### ISSUE TITLE
Continue with the project?

### ISSUE THREAD

**Issue by tatundria** [jharrels/scummy#47]
Please, this is the best ScummVM Frontend I have ever seen. I use it everyday. Can you continue with it? Or at least please help me to fix launch Indiana Jones & the Fate of Atlantis which is the only game that is not working with Scummy (I think it is a problem with the id "atlantis")

**Comment by jharrels** [comment]
> Please, this is the best ScummVM Frontend I have ever seen. I use it everyday. Can you continue with it? Or at least please help me to fix launch Indiana Jones & the Fate of Atlantis which is the only game that is not working with Scummy (I think it is a problem with the id "atlantis")

Howdy. I've been thinking about restarting the project, actually. I kept thinking ScummVM was going to introduce their own thumbnail front end, but I guess not. I'll see what I can do to get things started again. I no longer have a Mac in order to build the Mac version.

**Comment by tatundria** [comment]
Thank you for your reply! It would be great if you actually restart the project!
I have no idea on how to compile for Mac, but maybe with a little help from you I can make it.

Regarding the Indiana Jones and the Fate of Atlantis error, I was asking for help in discord and someone told me the following: “…There's now a conflict between `scumm:atlantis` and `cryomni3d:atlantis`(Atlantis: The Lost Tales) affecting some defunct frontends that don't specify the games as well as they could (or use --auto-detect). Either fix the JS code, or compile ScummVM without cryomni3d engine per Wiki instructions…”

Also, it would be great if you could add the game The Longest Journey to the Frontend, since it is now compatible with ScummVM.

Please, let me know if I can help you with anything.

**Comment by jharrels** [comment]
> Thank you for your reply! It would be great if you actually restart the project! I have no idea on how to compile for Mac, but maybe with a little help from you I can make it.

It requires installing Node.js on the Mac, downloading the source for Scummy and running "npm run build:mac". So it's pretty straightforward, if you want to give it a go. I can add you to the project as a contributor. I'll also see what other alternatives there are out there.

> Regarding the Indiana Jones and the Fate of Atlantis error, I was asking for help in discord and someone told me the following: “…There's now a conflict between `scumm:atlantis` and `cryomni3d:atlantis`(Atlantis: The Lost Tales) affecting some defunct frontends that don't specify the games as well as they could (or use --auto-detect). Either fix the JS code, or compile ScummVM without cryomni3d engine per Wiki instructions…”

I'll check this out. Looking at the code that snags the name of the game from scummvm, sure enough it's only getting whatever is on the right hand side of the colon. So, if it comes back with "scumm:atlantis" all it really gets is "atlantis", which must be blowing it up. Almost everything relies on that short name, though, so I'll have to put some thought into it before I try to fix it.

> Also, it would be great if you could add the game The Longest Journey to the Frontend, since it is now compatible with ScummVM.

Yep, I'll probably try to add everything that was added since... oh whatever version was out three years ago.

**Comment by tatundria** [comment]
Sure, I'll be glad to be a contributor and help on whatever I can :)

This is a great frontend that deserves to be updated.

**Comment by tatundria** [comment]
Hello again. Any news on this? Are you finally coming back? Sorry to bother!

**Comment by jharrels** [comment]
@tatundria It'll be a minute before I can devote much time to this. Apologies. I'll keep you posted.

**Comment by tatundria** [comment]
Hi @jharrels it`s been quite a while. Sorry for asking, are you still planning to continue with the project?

**Comment by jharrels** [comment]
Hi there. I guess it's only fair to let you know what's going on. My mother passed away in 2024, suddenly. After that, we found out she'd been covering up my father's young-onset Alzheimer's disease. He rapidly declined after her death until one day he fell and broke his hip, his arm, and some ribs. He never recovered from that. He passed away just two weeks ago.

I will eventually get back to this, but I've really got to clear my plate first before I can really do anything with it.

**Comment by tatundria** [comment]
Dear @jharrels , I'm really sorry to hear that. That’s a lot to go through, and I can’t imagine how hard it must be. Take all the time you need—no rush at all. If there’s anything I can do, just let me know. Sending you my deepest condolences.

**Comment by jharrels** [comment]
@tatundria Thank you. That's very kind of you. I took a month from work, got everything as settled as it can be.

I think it's time I dive back into this.

It's been a really long time since I've looked at the code and ScummVM. Over the years, I've had some new ideas and I think a good portion of it Scummy needs to be reworked. The one thing that I think worked well, was the way the games were displayed, but everything else was a little rough.

A lot has changed since I first wrote Scummy: I no longer have a Mac and my daily driver is now a Windows PC. Unfortunately, I don't have any way to build Mac executables without a Mac (or a VM or a cloud service). I also haven't programmed in Node.js in years, so I'm a bit rusty. I don't recall how I built the apps, either, so hopefully I can figure that out again.

I was thinking of distributing scummvm with it or perhaps downloading it from scummvm site automatically. Dunno yet, I have thoughts around this.

**Comment by tatundria** [comment]
@jharrels Glad to hear that you are back! Hope this process of updating Scummy is not so rough on you and we can enjoy a new version with bugfixes and new features, since Scummy is a great app with a muuuch better inferface than Scummvm itself. Please, let me know if I can help you with anything. Thanks a lot!!

**Comment by jharrels** [comment]
Thank you. I will let you know. So far, I have rewritten the piece that extracts all the engines and maps the games to their respective categories. As you can see here, it now has the full name instead of just "atlantis" which is the origin of the whole problem. So, that's one step.

`gameData={"scumm:atlantis":{"engine":"scumm","category":"lucasarts"},"scumm:indy3":{"engine":"scumm",".......`

**Comment by jharrels** [comment]
Scummy (or SAGE as I'm calling it now) is now getting the installed games properly.

`scumm:atlantis:
name: "Indiana Jones and the Fate of Atlantis"
versions: Array(3)
0: {version: "FM-TOWNS/English", versionShortName: "atlantis-fm"}
1: {version: "Floppy/DOS/English", versionShortName: "atlantis"}
2: {version: "CD/DOS/English", versionShortName: "atlantis-1"}
length: 3
__proto__: Array(0)
__proto__: Object
__proto__: Object`

Unfortunately, none of the rest of the code expects the "engine:gameid" format, so I've got to figure out where it's used and update it.

**Comment by tatundria** [comment]
Good news! That was actually the main issue that I found in the latest version. The other one is that the new games are missing. Thanks a lot

**Comment by jharrels** [comment]
Yep, there's quite a few new ones. I've updated everything so all games should show up as of version 2.9.0.

**Comment by jharrels** [comment]
![Image](https://github.com/user-attachments/assets/1767124a-30f2-444e-9aac-4ed2a902fea4)

I've got it quasi-working. Added the art for 11th Hour and Clandestiny, which were new since the last version of SAGE/Scummy. Can't see it here, but there's a display issue for favorites that I've yet to figure out. I'll have to rename every single box art that's in the app, which I'm not sure how to automate just yet. It's progress, though!

**Comment by jharrels** [comment]
@tatundria I just released 1.1.0. Give it a try and let me know how it goes! 
https://github.com/jharrels/SAGE/releases/tag/1.1.0

**Comment by tatundria** [comment]
> [@tatundria](https://github.com/tatundria) I just released 1.1.0. Give it a try and let me know how it goes! https://github.com/jharrels/SAGE/releases/tag/1.1.0

That is GREAT to hear! I am on holidays until next Wednesday, but as soon as I arrive at home I will give it a try and will give you my feedback 😁 thanks a LOT!

Also, I have a Mac, so I will test it on a Windows Virtual Machine. How can I compile it for Mac?

**Comment by jharrels** [comment]
Awesome! Enjoy your holiday. When you get back, drop me a line at jmharrels@gmail.com and we can chat about the Mac version.
